src/bin/common/parse_command.ml

```diff
@@ -1230,33 +1210,45 @@ let parse_theory_opt =
 let parse_fmt_opt =
 
   let docs = s_fmt in
+  let docv = "CHANNEL" in
 
-  let std_formatter =
-    let doc = Format.sprintf
-        "Set the standard formatter used by default to output the results,
+  let std_output, deprecated_std_output =
+    let doc =
+      Format.sprintf
+        "Set the standard output used by default to print the results,
     models and unsat cores. Possible values are %s."
-        (Arg.doc_alts ["stdout"; "stderr"; "<filename>"]) in
-    Arg.(value & opt formatter_conv Stdout & info ["std-formatter"] ~docs ~doc)
+        (Arg.doc_alts ["stdout"; "stderr"; "<filename>"])
+    in
+    let deprecated = "this option is depreciated. Please use --std-output." in
+    Arg.(value & opt string "stdout" & info ["std-output"] ~docv ~docs ~doc),
+    Arg.(value & opt string "stdout" & info ~deprecated ["std-formatter"]
```

Comment:
I would prefer something like this:

```ocaml
let merge_formatters default preferred deprecated =
  match preferred, deprecated with
  | Some fmt, _ -> fmt
  | None, Some fmt -> fmt
  | None, None -> default

let std_formatter = 
  ...
  let std_output = Arg.(value & opt (some' formatter_conv) None & info ["std-output"] ~docs ~doc ~docv in
  let std_formatter = Arg.(value & opt (some' formatter_conv) None & info ["std-formatter"] ~deprecated in
  Term.(const (merge_formatters Stdout) $ std_output $ std_formatter)
```

so that the existence of deprecated arguments stays within the CLI API.

Reply:
Looks better indeed :)